AngularApp/src/app/diagnostic-data/components/time-series-graph/time-series-graph.component.ts

```diff
@@ -115,7 +117,10 @@ export class TimeSeriesGraphComponent extends DataRenderBaseComponent implements
     if (data.table.rows.length <= 1) {
       this.timeGrain = this._getNoDataTimegrain();
     }
-    this._prepareStartAndEndTime();
+
+    if (!this.customizeXAxis) {
```

Comment:
I don't love all these if statements as this can lead to buggy code or make it hard to change later. Is there a way to reorganize this method so there can be only one if switch that will handle the logic for the two modes?